drivers/serial/uart_nrfx_uarte.c

```diff
@@ -189,6 +188,48 @@ static inline NRF_UARTE_Type *get_uarte_instance(const struct device *dev)
 	return config->uarte_regs;
 }
 
+#ifndef CONFIG_PINCTRL
+static void uarte_nrfx_pins_configure(const struct device *dev, bool sleep)
+{
+	const struct uarte_nrfx_config *cfg = get_dev_config(dev);
+
+	if (!sleep) {
+		nrf_gpio_pin_write(cfg->tx_pin, 1);
+		nrf_gpio_cfg_output(cfg->tx_pin);
```

Comment:
This assumes that `tx_pin` has a valid value (the `tx-pin` property was required so far), but currently it can be `NRF_UARTE_PSEL_DISCONNECTED`. Similar story in `uart_nrfx_uart.c`.

Reply:
guarded against disconnected possibility in both uart/e drivers.